resharper/src/resharper-unity/Utils/UnityComponentRelatedExpressionFinder.cs

```diff
@@ -0,0 +1,372 @@
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+using QualifierEqualityComparer = JetBrains.ReSharper.Psi.CSharp.Impl.ControlFlow.ControlFlowWeakVariableInfo.QualifierEqualityComparer;
```

Comment:
I believe that it'd be better to write specialized equality comparer instead of using an existing one from `ControlFlowWeakVariableInfo` as it wasn't supposed to be used outside of control flow inspections.

1. It does more work than needed for these inspections e.g. processing indexers in qualifiers.
2. It might change at any moment according to it's intended usages in control flow inspections with no regards of such external dependencies since they aren't supposed to be there.
3. It does not provide options for customizing behavior e.g. how to handle conditional accesses.

It looks like checking pairs of reference expressions one by one is a simple enough case to implement it separately.

Also please